packages/dtslint/src/checks.ts

```diff
@@ -130,3 +132,33 @@ export function checkTsconfig(dirPath: string, options: CompilerOptionsRaw): str
   }
   return errors;
 }
+
+export interface AttwResult {
+  status: "pass" | "fail" | "error";
+  output: string;
+}
+
+export function runAreTheTypesWrong(dirPath: string, configPath: string): AttwResult {
+  const packageJsonContent = readJson(joinPaths(dirPath, "package.json"));
+  const mangledName = packageJsonContent.name.replace(/^@types\//, "");
+  const unmangledName = unmangleScopedPackage(mangledName) || mangledName;
+  const tarballName = `types-${mangledName}-${packageJsonContent.version}.tgz`;
+  execSync("npm pack", { cwd: dirPath, stdio: "ignore" });
+  try {
+    const output = execSync(`npx attw -p ${unmangledName} --definitely-typed ${tarballName} --config-path ${configPath}`, {
```

Comment:
This `npx` is likely to fail if someone has `corepack` enabled (which is generally recommended) since `npx` is tied to the wrong package manager; can we dep on `attw` and then `require.resolve` or just hardcode a path to the bin instead?

Reply:
I don’t think there’s a `require.resolve` equivalent for `bin` files. Can we count on `dtslint/node_modules/.bin/attw` or will that possibly be hoisted?

Reply:
Never mind, I can `require.resolve(@'arethetypeswrong/cli/package.json')` since there are no `exports`

Reply:
Yes, I just failed to notice it...

Reply:
For that one, you can just use `COREPACK_ENABLE_STRICT=0` (there are some examples I used previously), or we can skip calling `npm` and just ask `pacote`, which has the ability to produce tarballs.